cpp_linter/rest_api/github_api.py

```diff
@@ -446,6 +450,51 @@ def post_review(
                 summary_only=summary_only,
                 review_comments=review_comments,
             )
+        ignored_reviews = []
+        if not summary_only:
+            found_threads = self._get_existing_review_comments(
+                no_dismissed=reuse_comments and not delete_comments)
+            if found_threads:
+                if reuse_comments:
+                    # Keep already posted comments if they match new ones
+                    review_comments_suggestions = review_comments.suggestions
+                    review_comments.suggestions = []
+                    existing_review_comments = []
+                    for thread in found_threads:
+                        for comment in thread["comments"]["nodes"]:
+                            found = False
+                            line_start = comment["originalStartLine"] if comment["originalStartLine"] is not None else -1
+                            line_end = comment["originalLine"]
+                            if comment["startLine"] is not None:
+                                line_start = comment["startLine"] if comment["startLine"] is not None else (
+                                    comment["line"] if comment["line"] is not None else line_start)
+                                line_end = comment["line"] if comment["line"] is not None else line_end
```

Comment:
I understand that after all that `... if ... else ...`, this is basically the same, yes?
```suggestion
                            if "line" in comment and comment["line"] is not None:
                                line_end = comment["line"]
                                line_start = line_end
                                if "startLine" in comment and comment["startLine"] is not None:
                                    line_start = comment["startLine"]
```

Reply:
No not quite. Theres a few combinations of things that can happen here. Firstly there are the original lines ("originalLine" and "originalStartLine") which are the defaults unless a comment has moved due to line changes in which case "startLine" and "line" are used. However single line comments dont have both as only multi line comments have both. So for instance `line_end` should be "line" if it exists, "line" is only valid if theres a "startLine", if neither then it should be "originalLine"
I did a bunch of tests on a repo with single and multi-line comments that were new/moved due to changes and the existing logic seemed to be the one that was correct for all those cases